### DISCUSSION TITLE
[4.0] [Atum] [UX] Wrong element's visual priority in Users -> Manage

### DISCUSSION THREAD

**Discussion by simbus82** [joomla/joomla-cms#47135] [2. Cold Features]
### Steps to reproduce the issue
Go to Users -> Manage

### Expected result
Username or Name are the more visually relevant elements. Click on the name (or username) link is easy.

### Actual result
![image](https://user-images.githubusercontent.com/1375730/117645515-fb9e7080-b18a-11eb-9372-3de68a4ad111.png)

1. There are less than 2 pixel distance from name link and "add note" link area.
Name link doesn't have a sufficient click area. There's no link for username.
2. "add note" button steal all of visual priority of elements in this UI.
"add note" is absolutely the less relevant (less used and less important for users management processes) element in this page.

![image](https://user-images.githubusercontent.com/1375730/117647351-27225a80-b18d-11eb-9676-ad66fb80b260.png)
Page show a really wrong "information": "add note" is the main actor of the page.

### System information (as much as possible)
Latest 4.0 build

### Additional comments

Some users list good UX examples:
https://dribbble.com/shots/7046861-Network-Management-Dashboard-UI-Design/attachments/47229?mode=media
https://dribbble.com/shots/6158106-Members-List-design-concept

**Comment by brianteeman** [comment]
> "add note" is absolutely the less relevant (less used and less important for users management processes) element in this page.

That's subjective and based on your experience of using the site. It doesn't have to be the same for everyone. For example I use notes very very heavily on one site.

Not going to disagree that the ui can be improved though

**Comment by PhilETaylor** [comment]
"Add Note" has always had really bad UI :-( 

I suggest 

 - Remove the big blue button, and replace with a list select, and add a TOOLBAR button in the actions menu for Add Note
 - When a user HAS a note, have a new small icon next to their name (maybe right aligned) that serves as the hover and click point to view the note links, (Display note, list notes)

**Comment by PhilETaylor** [comment]
> For example I use notes very very heavily on one site.

Then you should be using the User Notes extension, and not the Users extension for that, maybe the User Notes extension needs a better UI than just a list view so that it becomes more useable as the main extension to use for, er, User Notes instead  of having its finger stuck into the Users extension in this page. 

The big blue button is not in keeping really with any other view in Joomla

**Comment by chmst** [comment]
Just for information: In GSoC2021 feature enhancement, it is intended to make user notes as optional feature. This includes better UX here.

**Comment by PhilETaylor** [comment]
Optional? how?

**Comment by brianteeman** [comment]
big mistake.  just because some people dont use it because they dont understand it

**Comment by brianteeman** [comment]
@simbus82 How about moving the note part to a column - similar to the permissions column?

**Comment by PhilETaylor** [comment]
Nooooo there simply is not enough room in this table for yet another column (and its padding), especially when 2fa is enabled :) 

Already the username only gets about 5 chars room sometimes :-(

**Comment by brianteeman** [comment]
was just a suggestion

**Comment by simbus82** [comment]
> @simbus82 How about moving the note part to a column - similar to the permissions column?

Something like this could be nice, but probably @PhilETaylor is right about space.
![image](https://user-images.githubusercontent.com/1375730/117700628-8f416280-b1c6-11eb-807b-cd99fc90e7dc.png)

Or, imho, could be very useful to put these "tools" in a 3-point menu, i don't know about the accessibility, but it's used everywhere in big companies app too.
This is a good UI devolopment method that give us more freedom to add more contextual "tools and action" in the future.
![image](https://user-images.githubusercontent.com/1375730/117701238-4938ce80-b1c7-11eb-8d6d-9fa94e462531.png)
Here we can add many functions like "add a note", "send a private message", etc.

**Comment by brianteeman** [comment]
Sorry I must have not been thinking correctly. My use of user notes is all done in the user notes component and not the users component

**Comment by eopws** [comment]
Maybe add context menu which opens with the right mouse button and move the add note button there? Just suggestion.

**Comment by C-Lodder** [comment]
Or just move it under the "Actions" toolbar dropdown, like every other action?

![Untitled](https://user-images.githubusercontent.com/2019801/117931186-c9aa1d00-b2f6-11eb-9c2d-3ac4beab5e3a.png)

**Comment by simbus82** [comment]
> Or just move it under the "Actions" toolbar dropdown, like every other action?
> 

For me ... it is the best solution!

**Comment by brianteeman** [comment]
Unfortunately that would be a loss of functionality.
![image](https://user-images.githubusercontent.com/1296369/117943357-f57fcf80-b303-11eb-9199-0c9b0936f509.png)

The main being that you can easily see if there is a note for a user

**Comment by brianteeman** [comment]
How about a combination of the action dropdown as suggested by @C-Lodder and an icon(with link) to indicate if a note exists for a user

![image](https://user-images.githubusercontent.com/1296369/117954290-cb7fda80-b30e-11eb-898f-41d8128a579d.png)

**Comment by PhilETaylor** [comment]
> How about a combination of the action dropdown as suggested by @C-Lodder and an icon(with link) to indicate if a note exists for a user

er... tap tap, is this thing on, am I talking to myself here? hahaha

![Image 2021-05-12 at 10 44 30 am](https://user-images.githubusercontent.com/400092/117954571-0eda4900-b30f-11eb-873e-ccbb0543146e.jpg)

**Comment by simbus82** [comment]
I'm not familiar with the admin backend code to fix those functions related to UI, I don't think I can do a good PR :-(

**Comment by Quy** [comment]
Related #29509

**Comment by simbus82** [comment]
All the admins of our CMSs (customers above all) are asking me to remove that button from the user management (and permissions) from their 4.x websites because it is "literally unwatchable".
I am acting brutally with css to hide it (practically nobody uses the notes system) but obviously it is not a solution.
In Joomla 4.1 there is still this huge UI flaw.
Also the "Permission" button has the same problem.
![image](https://user-images.githubusercontent.com/1375730/152760069-180782a5-7ead-4e59-a6c1-ef2d9417e5b6.png)

There really isn't a CMS where is not used solutions like the Clodder/Brian/Phil proposal or better yet a context menu, or an hamburger at the end or at the beginning of the row. Never see so may buttons with label in the row of a list, it's discouraged from any UX perspective.

Example:
![image](https://user-images.githubusercontent.com/1375730/152761472-2525baea-3fd6-4b74-875e-83b34a5de3d6.png)

I hope this ugliness can be solved before Joomla 4.1 is released.
As I said, I can't do a PR for this problem, I don't have the skills.
We can't let the project stumble upon on these things :-(

**Comment by brianteeman** [comment]
> I hope this ugliness can be solved before Joomla 4.1 is released.

Unfortunately not

**Comment by simbus82** [comment]
> > I hope this ugliness can be solved before Joomla 4.1 is released.
> 
> Unfortunately not

Joomla 4.3?

**Comment by simbus82** [comment]
@crystalenka you are great in UX, how can we correct this "crap" UI?

**Comment by crystalenka** [comment]
I'll add it to my list of things to look at. :)

**Comment by chmst** [comment]
> @crystalenka you are great in UX, how can we correct this "crap" UI?

@simbus82 you can join the team and work for it.
What about making user notes as optional and add it on request like workflow? I agree that this button at this place is not a good solution.

**Comment by simbus82** [comment]
> I am acting brutally with css to hide it (practically nobody uses the notes system) but obviously it is not a solution. 
> As I said, I can't do a PR for this problem, I don't have the skills.

@chmst i can only give advice (or can I tell my idea) on the User/Customer eXperience.
For example in many years and many hundreds of professional joomla website created, no administrators/managers have ever used the notes function once, so for me "notes" it's a removable function or manageable it like an installable plugin.
But the real problem is the UI and the visual priority of items in the page. That button is really visually wrong, I don't mind the function.

Apart from that, with Joomla 4 there is too much JS and too many compiled CSS for my code development skills.
So I can't be very helpful ...
